gpt_index/readers/slack.py

```diff
@@ -2,39 +2,55 @@
 import logging
 import os
 import time
+from datetime import datetime
 from typing import List, Optional
 
-from gpt_index.readers.base import BaseReader
-from gpt_index.readers.schema.base import Document
+from llama_index import Document
+from llama_index.readers.base import BaseReader
+
+logger = logging.getLogger(__name__)
 
 
 class SlackReader(BaseReader):
     """Slack reader.
 
-    Reads conversations from channels.
+    Reads conversations from channels. If an earliest_date is provided, an
+    optional latest_date can also be provided. If no latest_date is provided,
+    we assume the latest date is the current timestamp.
 
     Args:
         slack_token (Optional[str]): Slack token. If not provided, we
             assume the environment variable `SLACK_BOT_TOKEN` is set.
-
+        earliest_date (Optional[datetime]): Earliest date from which
+            to read conversations. If not provided, we read all messages.
+        latest_date (Optional[datetime]): Latest date from which to
+            read conversations. If not provided, defaults to current timestamp
+            in combination with earliest_date.
     """
 
-    def __init__(self, slack_token: Optional[str] = None) -> None:
+    def __init__(
+        self,
+        slack_token: Optional[str] = None,
+        earliest_date: Optional[datetime] = None,
+        latest_date: Optional[datetime] = None,
+    ) -> None:
         """Initialize with parameters."""
-        try:
-            from slack_sdk import WebClient
-        except ImportError:
-            raise ImportError(
-                "`slack_sdk` package not found, please run `pip install slack_sdk`"
-            )
+        from slack_sdk import WebClient
+
         if slack_token is None:
             slack_token = os.environ["SLACK_BOT_TOKEN"]
-            if slack_token is None:
-                raise ValueError(
-                    "Must specify `slack_token` or set environment "
-                    "variable `SLACK_BOT_TOKEN`."
-                )
+        if slack_token is None:
+            raise ValueError(
+                "Must specify `slack_token` or set environment "
+                "variable `SLACK_BOT_TOKEN`."
+            )
         self.client = WebClient(token=slack_token)
+        if earliest_date is not None:
```

Comment:
would be good to add some validation (raise error if latest date is specified but earliest_date is not)

Reply:
Good point. Added/done.